apps/spruce/src/components/TaskStatusIconLegend/index.tsx

```diff
@@ -1,109 +1,152 @@
-import { useState } from "react";
+import { useRef, useState } from "react";
 import styled from "@emotion/styled";
 import IconButton from "@leafygreen-ui/icon-button";
-import Popover from "@leafygreen-ui/popover";
-import { Disclaimer, Overline } from "@leafygreen-ui/typography";
+import Popover, { Align, Justify } from "@leafygreen-ui/popover";
+import { Body, Overline } from "@leafygreen-ui/typography";
 import { size, zIndex } from "@evg-ui/lib/constants/tokens";
+
+import { useWaterfallAnalytics } from "analytics";
 import { useProjectHealthAnalytics } from "analytics/projectHealth/useProjectHealthAnalytics";
 import Icon from "components/Icon";
 import { PopoverContainer } from "components/styles/Popover";
-import { groupedIconStatuses } from "components/TaskStatusIcon";
+import {
+  groupedIconStatuses,
+  waterfallGroupedStatuses,
+} from "components/TaskStatusIcon";
 import { taskStatusToCopy } from "constants/task";
+import { useOnClickOutside } from "hooks";
+
+type LegendContentProps = {
+  useWaterfall: boolean;
+};
 
-export const LegendContent = () => (
-  <Container>
-    {groupedIconStatuses.map(({ icon, statuses }) => {
-      const label = statuses.map((status) => (
-        <Disclaimer key={status}>{taskStatusToCopy[status]}</Disclaimer>
-      ));
-      return (
+export const LegendContent: React.FC<LegendContentProps> = ({
+  useWaterfall,
+}) => {
+  const Container = useWaterfall
+    ? WaterfallContainer
+    : MainlineCommitsContainer;
+
+  const groupedStatuses = useWaterfall
+    ? waterfallGroupedStatuses
+    : groupedIconStatuses;
```

Comment:
Hm this is supposed to show the old icon set on mainline commits right? That isn't working for some reason (and I do think showing separate icon sets is the desired behavior)
<img width="1507" alt="image" src="https://github.com/user-attachments/assets/09f26626-124d-4feb-ab90-abde0fdaf195">

Reply:
added more boolean logic to make this behave better !